git commit --amend does not work when tests_lines exists

Constellation opened the pull request:
#### 6675efa0b52b2faac4cd523b3e29faf9ed5e00fc
<pre>
git commit --amend does not work when tests_lines exists
<a href="https://bugs.webkit.org/show_bug.cgi?id=298813">https://bugs.webkit.org/show_bug.cgi?id=298813</a>
<a href="https://rdar.apple.com/160519511">rdar://160519511</a>

Reviewed by Jonathan Bedard.

After 295625@main, tests_lines for amend message is generated
via [commit_message_parser.tests_lines, &apos;&apos;]. But this generates list
of list and it is rejected by &apos;\n&apos;.join(list_of_list). As a result,
precommit-hook fails and `git commit --amend` does not work.
This patch fixes it by changing it to list concatenation.

* Tools/Scripts/hooks/prepare-commit-msg:

Canonical link: <a href="https://commits.webkit.org/299929@main">https://commits.webkit.org/299929@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/71fd722bd84b8b485ea116c758a09c7e212853d9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/120754 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/40448 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/31100 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/127151 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/72829 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/1702997a-2058-4568-943e-6c28815e22b8) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/122630 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/41145 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/49025 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/91720 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/60966 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/03b77fdc-7cf6-4cd7-9158-30490028ef20) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/123706 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/32851 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/108237 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/72268 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/bd1a723f-1bc5-43ba-9711-d16f03d9fc0d) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/31881 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/26350 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/70754 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/102341 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/26527 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/130015 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/47675 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/36199 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/100334 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/48043 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/104420 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/100173 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/45622 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/23660 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/44338 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19164 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/47537 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/47006 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/50352 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/48692 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0bd6acccd569f043ac77fac54738ee8c8d103cd4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50700

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/71fd722bd84b8b485ea116c758a09c7e212853d9. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50700

Constellation committed:
git commit --amend does not work when tests_lines exists
https://bugs.webkit.org/show_bug.cgi?id=298813
rdar://160519511

Reviewed by Jonathan Bedard.

After 295625@main, tests_lines for amend message is generated
via [commit_message_parser.tests_lines, '']. But this generates list
of list and it is rejected by '\n'.join(list_of_list). As a result,
precommit-hook fails and `git commit --amend` does not work.
This patch fixes it by changing it to list concatenation.

* Tools/Scripts/hooks/prepare-commit-msg:

Canonical link: https://commits.webkit.org/299929@main

webkit-commit-queue commented:
Committed 299929@main (6675efa0b52b): <https://commits.webkit.org/299929@main>

Reviewed commits have been landed. Closing PR #50700 and removing active labels.
